mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -353,41 +353,6 @@ class ConstructorPattern final
   }
 };
 
-//===----------------------------------------------------------------------===//
```

Comment:
I see that you moved this code to ConvertToLLVMABI pass, as it needs to be before ConvertToLLVMABI conversions. But these doesn't really belong to the logic of memref -> pointer. Do you think it makes sense to have a SYCLToSYCL pass for these kinds of conversions. And add that in the pipeline before ConvertToLLVMABI pass.

Reply:
Yes, it does make sense. I was thinking on doing that on another PR if you are OK with that.

Reply:
I am OK with another PR.